concent_api/core/tests/test_integration_force_payment.py

```diff
@@ -181,6 +183,69 @@ def test_provider_send_force_payment_with_subtask_results_accepted_where_ethereu
         )
         self._assert_stored_message_counter_not_increased()
 
+    def test_that_if_provider_sends_force_payment_with_two_different_providers_ethereum_accounts_concent_should_refuse(self):
+        """
+        Expected message exchange:
+        Provider  -> Concent:    ForcePayment
+        Concent   -> Provider:   ServiceRefused
+        """
+        report_computed_task1 = self._get_deserialized_report_computed_task(
+            timestamp="2018-02-05 10:00:05",
+            task_to_compute=self._get_deserialized_task_to_compute(
+                timestamp="2018-02-05 10:00:00",
+                deadline="2018-02-05 10:00:10",
+                subtask_id=self._get_uuid('1'),
+                price=15000,
+            ))
+
+        provider1_keys = cryptography.ECCx(None)
+        provider_ethereum_public_key = encode_hex(provider1_keys.raw_pubkey)
+
+        report_computed_task2 = self._get_deserialized_report_computed_task(
+            timestamp="2018-02-05 10:00:05",
+            task_to_compute=self._get_deserialized_task_to_compute(
+                timestamp="2018-02-05 10:00:00",
+                deadline="2018-02-05 10:00:10",
+                subtask_id=self._get_uuid('2'),
+                provider_ethereum_public_key=provider_ethereum_public_key,
+                price=7000,
+            ))
+
+        subtask_results_accepted_list = [
+            self._get_deserialized_subtask_results_accepted(
+                timestamp="2018-02-05 10:00:15",
+                payment_ts="2018-02-05 11:55:00",
+                report_computed_task=report_computed_task1
+            ),
+            self._get_deserialized_subtask_results_accepted(
+                timestamp="2018-02-05 9:00:15",
+                payment_ts="2018-02-05 11:55:00",
+                report_computed_task=report_computed_task2
+            ),
+        ]
+
+        serialized_force_payment = self._get_serialized_force_payment(
+            timestamp="2018-02-05 12:00:20",
+            subtask_results_accepted_list=subtask_results_accepted_list
+        )
+
+        with freeze_time("2018-02-05 12:00:20"):
+            with mock.patch(
+                'core.message_handlers.bankster.settle_overdue_acceptances',
+                side_effect=self.settle_overdue_acceptances_mock
+            ):
+                response = self.send_request(
+                    url='core:send',
+                    data=serialized_force_payment,
+                )
+
+        self._test_response(
+            response,
+            status=200,
+            key=self.PROVIDER_PRIVATE_KEY,
+            message_type=message.concents.ServiceRefused,
```

Comment:
`reason` should be checked too if available.

Reply:
done